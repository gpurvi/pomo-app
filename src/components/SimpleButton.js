import React from 'react';

const SimpleButton = (props) => (
    <button data-attr={props.dataAttr} onClick={props.onClick}>{props.text}</button>
);

export default SimpleButton;