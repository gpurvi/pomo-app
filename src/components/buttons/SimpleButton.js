import React from 'react';

const SimpleButton = (props) => (
    <button
        className='btn btn-secondary btn-sm'
        type='button'
        disabled={props.disabled}
        data-attr={props.dataAttr}
        onClick={props.onClick}
    >
        {props.text}
    </button>
);

export default SimpleButton;