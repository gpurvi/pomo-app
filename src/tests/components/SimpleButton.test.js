import React from 'react';
import {shallow} from 'enzyme';
import SimpleButton from '../../components/buttons/SimpleButton';

test('renders SimpleButton with text', () => {
    const wrapper = shallow(<SimpleButton text="text"/>);
    expect(wrapper).toMatchSnapshot();
});

test('renders SimpleButton with text passed as prop', () => {
    const wrapper = shallow(<SimpleButton text="text"/>);
    expect(wrapper.text()).toBe('text');
});

test('should call onClick', () => {
    const onClick = jest.fn();
    const wrapper = shallow(<SimpleButton onClick={onClick}/>);
    wrapper.simulate('click');
    expect(onClick).toHaveBeenCalled();
});