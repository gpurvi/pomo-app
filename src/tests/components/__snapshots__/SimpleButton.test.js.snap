// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders SimpleButton with text 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SimpleButton
    text="text"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "text",
      "data-attr": undefined,
      "onClick": undefined,
    },
    "ref": null,
    "rendered": "text",
    "type": "button",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "text",
        "data-attr": undefined,
        "onClick": undefined,
      },
      "ref": null,
      "rendered": "text",
      "type": "button",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
